ui/redux.md

```diff
@@ -0,0 +1,120 @@
+### Redux
+In UI-classic we enabled [redux](https://github.com/reactjs/redux) to work with state of application so we can easilly
+ detect from where which action comes and to see the flow of data.
+
+#### Writing reducers
+To change state of application you will need to write reducer which is basically simple function which takes current 
+state and action. Result of such reducer is somehow changed state based on action.
+
+Action consists of
+ * Type of action - string which says what action has been triggered
+ * Payload - usually JS object with some data, which will be applied to state
+
+##### Simple reducer example
+We are using redux in slightly different way, so we can control which reducer is active at given time. To create new
+reducer first you have to create it and register it into redux.
+
+1. Breaking reducers into simple functions, might be a bit confusing because it doesn't use switches as any redux example
+however it's easier to test and maintain.
+
+```javascript 1.8
```

Comment:
Is this javascript or typescript, syntax is confusing...

Reply:
It's javascript, but I forgot to remove types from couple of functions. Good point.